packages/cli/src/runDeploy.ts

```diff
@@ -52,14 +56,13 @@ export const deployOptions = {
 
 export type DeployOptions = InferredOptionTypes<typeof deployOptions>;
 
-/**
- * Given some CLI arguments, finds and resolves a MUD config, foundry profile, and runs a deploy.
- * This is used by the deploy, test, and dev-contracts CLI commands.
- */
 export async function runDeploy(opts: DeployOptions): Promise<WorldDeploy> {
-  const salt = opts.salt;
-  if (salt != null && !isHex(salt)) {
-    throw new MUDError("Expected hex string for salt");
+  let salt: `0x${string}` | undefined = opts.salt as `0x${string}` | undefined;
+
+  if (salt != null) {
+    if (!isHex(salt)) {
+      salt = encodeToHex(salt) as `0x${string}`;
+    }
```

Comment:
we can simplify this to `const salt = opts.salt && !isHex(opts.salt) ? toHex(opts.salt) : opts.salt;`